presto-main/src/main/java/io/prestosql/sql/analyzer/Analyzer.java

```diff
@@ -84,14 +87,24 @@ public Analysis analyze(Statement statement, boolean isDescribe)
 
         // check column access permissions for each table
         analysis.getTableColumnReferences().forEach((accessControlInfo, tableColumnReferences) ->
-                tableColumnReferences.forEach((tableName, columns) ->
-                        accessControlInfo.getAccessControl().checkCanSelectFromColumns(
-                                accessControlInfo.getSecurityContext(session.getRequiredTransactionId(), session.getQueryId()),
-                                tableName,
-                                columns)));
+                tableColumnReferences.forEach((tableName, columns) -> {
+                    // TODO why getTableHandle sometimes gives empty()? (eg in TestAnalyzer.testExistingRecursiveView)
+                    metadata.getTableHandle(session, tableName).ifPresent(tableHandle ->
+                            accessControlInfo.getAccessControl().checkCanSelectFromColumns(
+                                    accessControlInfo.getSecurityContext(session.getRequiredTransactionId(), session.getQueryId()),
+                                    tableName,
+                                    columns.stream().flatMap(columnName -> getColumnMetadata(tableHandle, columnName).stream()).collect(toSet())));
+                }));
         return analysis;
     }
 
+    private Optional<ColumnMetadata> getColumnMetadata(TableHandle tableHandle, String columnName)
+    {
+        // TODO why sometimes .get(columnName) gives null? (eg in TestRowFilter.testView)
```

Comment:
I think we need to fix `TestRowFilter` to not to return null. Maybe it is using some custom connector or local query runner

Reply:
I have no idea how, somebody must help me